substrate/client/keystore/src/local.rs

```diff
@@ -391,6 +391,20 @@ impl Keystore for LocalKeystore {
 			self.sign::<ecdsa_bls377::Pair>(key_type, public, msg)
 		}
 
+			fn ecdsa_bls377_sign_with_keccak256(
```

Comment:
given the motivation for the joint ecdsa_bls keytypes is Ethereum, my expectation is that we'll predominantly be using the `sign_with_keccak256` flavor rather than default blake2 hashing.

Both for mental clarity and to limit developer mistakes, I tacitly suggest renaming `ecdsa_bls377_sign` to `ecdsa_bls377_sign_with_blake2` or to make `ecdsa_bls377_sign` generic as `ecdsa_bls377_sign<Hasher>`. This would put both hashers on equal footing and force developers to choose rather than mistakenly going for the default blake2 impl (which I expect to be required much less).

Reply:
> Another option could be a single ecdsa_bls377_sign with a mode (enum) param allowing to choose between: prehashed, blake2, keccak, ...

I feel this might result in wide spread changes across few modules, could I do it in another pull request?

Reply:
https://github.com/w3f/polkadot-sdk/issues/4